docs/Dialects/FIRRTL/FIRRTLAnnotations.md

```diff
@@ -1458,6 +1458,66 @@ Example:
 }
 ```
 
+### OutputDirAnnotation
+
+| Property   | Type   | Description                             |
+| ---------- | ------ | --------------------------------------- |
+| class      | string | `circt.OutputDirAnnotation`             |
+| dirname    | string | The output directory                    |
+| target     | string | Reference target                        |
+
+Specify the output directory for a module. The target must be a module, and must
+be local.
+
+Example:
+```json
+{
+  "class": "circt.OutputDirAnnotation",
+  "dirname": "verification",
+  "target": "~Foo|Bar"
+}
+```
+
+### OutputDirPrecedenceAnnotation
+
+| Property   | Type    | Description                             |
+| ---------- | ------- | --------------------------------------- |
+| class      | string  | `circt.OutputDirPrecedenceAnnotation`   |
+| name       | string  | The output directory                    |
+| parent     | string  | The parent output directory             |
+
+Specify the "parent" of an output directory.
+
+When Verilog is output, some modules will have user-specified output
+directories. When a module `M` is only instantiated by modules which have a
+common output directory `D`, we can also output module `M` in that same
+directory `D`.
+
+If the module M is instantiated under a set of directories `DS`, then the module
+`M` is placed in the output directory that is the least-common-ancestor (LCA) of
+the directories `DS`. The LCA is identified according to this output directory
+declaration annotation.
```

Comment:
This seems to be trying to tie directory output as specified to the use to instantiation structure.  Why?  If the user is specifying the directory, why not just use it?

Second, this is implicitly assuming that the compiler is responsible for the DUT/TB distinction.  Why are we interested in preserving this legacy SFC feature?